src/twister2/specification_processor.py

```diff
@@ -222,6 +227,12 @@ def should_skip_for_toolchain(test_spec: YamlTestSpecification, platform: Platfo
     if test_spec.toolchain_exclude and test_spec.toolchain_exclude & set(platform.toolchain):
         _log_test_skip(test_spec, platform, 'platform.toolchain in testcase.toolchain_exclude')
         return True
+    if (used_toolchain_version not in platform.toolchain) \
```

Comment:
IMO this section is more important than checking toolchain_allow and toolchain_exclude in test_cpec, so it can be moved to 224 line

Reply:
Changed as you suggested